ledger/internal/eval.go

```diff
@@ -1528,20 +1528,8 @@ transactionGroupLoop:
 			}
 
 			for _, br := range txgroup.balances {
-				// TODO: create cache of params as well
-				base.accounts[br.Addr] = ledgercore.ToAccountData(br.AccountData)
-				for aidx, val := range br.AccountData.AppParams {
-					base.appParams[ledgercore.AccountApp{Address: br.Addr, App: aidx}] = cachedAppParams{val, true}
-				}
-				for aidx, val := range br.AccountData.AssetParams {
-					base.assetParams[ledgercore.AccountAsset{Address: br.Addr, Asset: aidx}] = cachedAssetParams{val, true}
-				}
-				for aidx, val := range br.AccountData.AppLocalStates {
-					base.appLocalStates[ledgercore.AccountApp{Address: br.Addr, App: aidx}] = cachedAppLocalState{val, true}
-				}
-				for aidx, val := range br.AccountData.Assets {
-					base.assets[ledgercore.AccountAsset{Address: br.Addr, Asset: aidx}] = cachedAssetHolding{val, true}
-				}
+				base.accounts[br.Addr] = br.AccountData
+				// TODO: pull needed resources into cache?
```

Comment:
correct - but the hard part would be to modify loadAccounts so that it would load both the account, as well as the resources needed for the transaction group processing. This include the assetid ( for asset transactions ), application id ( for appcall transactions ) plus the foreign asset/app.